[Starting job container] | Blackhole post-commit tests (P100 nightly) | ttnn-unit-tests (P100a) / core ttnn unit test group [bh_p100]

## CI Failure Report

<!-- AUTO-TRIAGE-SUMMARY-START -->
**Summary:** [CI] Blackhole P100 ttnn-unit-tests failing: Docker pull of harbor.ci.tenstorrent.net fails due to DNS resolution error
<!-- AUTO-TRIAGE-SUMMARY-END -->

**Workflow:** Blackhole post-commit tests (P100 nightly)
**Job:** ttnn-unit-tests (P100a) / core ttnn unit test group [bh_p100]

### Regression timeline

| | Commit | Date | Run URL |
|---|---|---|---|
| Last passing | N/A | N/A | N/A |
| First failing | `b24ad48df9ce` | 2026-05-31T08:06:34Z | https://github.com/tenstorrent/tt-metal/actions/runs/26707277302 |
| Most recent failing | `10358b932eda` | 2026-06-14T08:25:41Z | https://github.com/tenstorrent/tt-metal/actions/runs/27493186485 |

The regression has persisted for over 2 weeks (since 2026-05-31).

### Failing job URLs (last 2 runs)

- https://github.com/tenstorrent/tt-metal/actions/runs/27493186485/job/81262804403
- https://github.com/tenstorrent/tt-metal/actions/runs/27461281596/job/81175960686

### Failing test path(s)

No test code is reached. The job fails during the **"Starting job container"** setup step before any tests run:

- Docker image: `harbor.ci.tenstorrent.net/ghcr.io/tenstorrent/tt-metal/tt-metalium/ubuntu-22.04-dev-amd64:7df58d9039d535d6df09c3b332fd9c32285113aa`

### Error excerpt (terminal failure from logs)

**Run 1** (runner: `tt-metal-ci-vm-215`, 2026-06-14):
```
Error response from daemon: Get "https://harbor.ci.tenstorrent.net/v2/": dial tcp: lookup harbor.ci.tenstorrent.net on 127.0.0.53:53: no such host
##[warning]Docker pull failed with exit code 1, back off 3.765 seconds before retry.
...
##[error]Docker pull failed with exit code 1
##[error]Value cannot be null. (Parameter 'ContainerId')
```

**Run 2** (runner: `tt-metal-ci-bm-yyzo-bh-gh02`, 2026-06-13):
```
Error response from daemon: Get "https://harbor.ci.tenstorrent.net/v2/": dial tcp: lookup harbor.ci.tenstorrent.net on 127.0.0.53:53: read udp 127.0.0.1:36453->127.0.0.53:53: i/o timeout
##[warning]Docker pull failed with exit code 1, back off 6.476 seconds before retry.
...
##[error]Docker pull failed with exit code 1
##[error]Value cannot be null. (Parameter 'ContainerId')
```

### Reproduction steps

This is an infrastructure failure and cannot be reproduced via a test command. To investigate:

1. On the affected runner(s), verify DNS resolution of `harbor.ci.tenstorrent.net`:
   ```bash
   nslookup harbor.ci.tenstorrent.net
   dig harbor.ci.tenstorrent.net
   ```
2. Verify network connectivity from the runner to the harbor registry:
   ```bash
   curl -v https://harbor.ci.tenstorrent.net/v2/
   ```
3. Check if the harbor registry service is up and its DNS record is correctly configured.

### Notes

- This is a **CI infrastructure failure**, not a test or code regression. The job never starts because the Docker image cannot be pulled.
- The failure affects at least two different runners (`tt-metal-ci-vm-215` and `tt-metal-ci-bm-yyzo-bh-gh02`), suggesting a registry-side or DNS infrastructure issue rather than a single machine problem.
- **Determinism check:**
  - Run 1 terminal error: `##[error]Docker pull failed with exit code 1` — DNS lookup returned `no such host` for `harbor.ci.tenstorrent.net`
  - Run 2 terminal error: `##[error]Docker pull failed with exit code 1` — DNS lookup timed out (`i/o timeout`) for `harbor.ci.tenstorrent.net`
  - Both runs share the same root cause (DNS cannot resolve `harbor.ci.tenstorrent.net`) and identical terminal error lines; the minor DNS sub-error variation (NXDOMAIN vs timeout) is consistent with a flaky/misconfigured registry DNS.

<!-- AUTO-TRIAGE-METADATA-START -->
`Auto-triage-workflow: Blackhole post-commit tests (P100 nightly)`
`Auto-triage-job-name: ttnn-unit-tests (P100a) / core ttnn unit test group [bh_p100]`
<!-- AUTO-TRIAGE-METADATA-END -->

--- a/.github/workflows/blackhole-post-commit.yaml
+++ b/.github/workflows/blackhole-post-commit.yaml
@@ -383,7 +383,7 @@ jobs:
       enable-watcher: ${{ inputs.enable-watcher || false }}
       enable-llk-asserts: ${{ inputs.enable-llk-asserts || false }}
   ttnn-unit-tests:
-    needs: [resolve-docker-pull-refs, resolve-artifacts, generate-matrix]
+    needs: [resolve-docker-pull-refs, resolve-artifacts, generate-matrix, check-harbor]
     if: ${{ always() && !cancelled() && (inputs.run-ttnn-unit-tests == true || (github.event_name == 'schedule' && needs.resolve-artifacts.outputs.schedule-runs-all-tests-effective == 'true')) }}
     secrets: inherit
     uses: ./.github/workflows/ttnn-post-commit.yaml

--- a/.github/workflows/blackhole-post-commit.yaml
+++ b/.github/workflows/blackhole-post-commit.yaml
@@ -376,7 +376,7 @@ jobs:
     with:
       arch: blackhole
       runner-label: ${{ matrix.test-group }}
-      timeout: 20
+      timeout: 30
       # CIv1 runners: cannot use harbor-prefix (resolve-docker-pull-refs)
       docker-image: ${{ needs.resolve-artifacts.outputs.dev-docker-image }}
       wheel-artifact-name: ${{ needs.resolve-artifacts.outputs.wheel-artifact-name }}
@@ -393,7 +393,8 @@ jobs:
         test-group: ${{ fromJson(needs.generate-matrix.outputs.civ2-matrix) }}
     with:
       enabled-skus: ${{ (matrix.test-group == 'P100' || matrix.test-group == 'P100a') && 'bh_p100' || 'bh_p150b_civ2' }}
-      docker-image: ${{ needs.resolve-docker-pull-refs.outputs.ci-test-docker-image }}
+      docker-image: ${{ needs.resolve-artifacts.outputs.ci-test-docker-image }}
+      harbor-prefix: ${{ needs.check-harbor.outputs.harbor-prefix }}
       wheel-artifact-name: ${{ needs.resolve-artifacts.outputs.wheel-artifact-name }}
       enable-watcher: ${{ inputs.enable-watcher || false }}
       timeout-minutes-override: ${{ inputs.enable-watcher && 120 || 0 }}

--- a/.github/workflows/umd-unit-tests.yaml
+++ b/.github/workflows/umd-unit-tests.yaml
@@ -34,8 +34,7 @@ jobs:
     name: UMD tests ${{ inputs.arch }} ${{ inputs.runner-label }}
     runs-on: >-
       ${{
-        (inputs.runner-label == 'P100a') && fromJSON(format('["in-service", "cloud-virtual-machine", "P100"]'))
-        || format('tt-ubuntu-2204-{0}-viommu-stable', inputs.runner-label)
+        format('tt-ubuntu-2204-{0}-viommu-stable', inputs.runner-label)
       }}
     container:
       image: ${{ inputs.docker-image }}

